### PR TITLE
🐛 fix: (helm/v1alpha1): add missing replicas field to the Deployment generated

### PR DESCRIPTION
**Pull request by monteiro-renato**
I've also moved the field out of "**_container_**" since I find it to be a bit less confusing from a k8s concepts point of view.

### CODE REVIEW

File: pkg/plugins/optional/helm/v1alpha/scaffolds/internal/templates/values.go [hunk 1]
```diff
@@ -57,11 +57,11 @@ func (f *HelmValues) SetTemplateDefaults() error {
 
 const helmValuesTemplate = `# [MANAGER]: Manager Deployment Configurations
 controllerManager:
+  replicas: 1
   container:
     image:
       repository: controller
-      tag: latest
-    replicas: 1
+      tag: latest
```
Comment (camilamacedo86) [inline pkg/plugins/optional/helm/v1alpha/scaffolds/internal/templates/values.go:64]: /ok-to-test

File: testdata/project-v4/go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module sigs.k8s.io/kubebuilder/testdata/project-v4
 go 1.22.0
 
 require (
-	github.com/cert-manager/cert-manager v1.16.1
+	github.com/cert-manager/cert-manager v1.16.2
```
Comment (camilamacedo86) [inline testdata/project-v4/go.mod:6]: this one are unrelated changes.

Comment (monteiro-renato) [reply testdata/project-v4/go.mod:6]: Yep, it was the same problem I saw here: https://github.com/kubernetes-sigs/kubebuilder/pull/4339#issuecomment-2489567278
Now that the PR mentioned in the comment has been merged I can rebase and this autogenerated changes will be gone :)

**Review by camilamacedo86 — APPROVED** [review]
Good catcher 🥇 
Thank you a lot for your help on those

/lgtm
/approve
/ok-to-test
